packages/components/src/utils/getParsedDate.ts

```diff
@@ -20,7 +21,14 @@ export const getParsedDate = (dateString: string) => {
 
       try {
         if (isMatch(dateString, format)) {
-          return parse(dateString, format, new Date())
+          let offsetDate = dateString
+          if (format === TIMEZONE_DATE_FORMAT) {
+            offsetDate = new Date(
+              // Add 8 hours to account for the Singapore timezone offset
+              new Date(dateString).getTime() + 8 * 60 * 60 * 1000,
```

Comment:
Doesn't `new Date()` by default get the date in local timezone?

Reply:
from what i understand, `new Date()` is the reference date that provides default values for any missing components in `dateString`. e.g. if the `dateString` includes only the month and day, the year is taken from the reference date